cmd/kubelet/app/server.go

```diff
@@ -316,13 +317,44 @@ func initConfigz(kc *componentconfig.KubeletConfiguration) (*configz.Config, err
 	return cz, err
 }
 
-// validateConfig validates configuration of Kubelet and returns an error if the input configuration is invalid.
-func validateConfig(s *options.KubeletServer) error {
+// validateConfig validates configuration of Kubelet and returns a slice of errors if the input configuration is invalid.
+func validateConfig(s *options.KubeletServer) []error {
+	var errors []error
 	if !s.CgroupsPerQOS && len(s.EnforceNodeAllocatable) > 0 {
-		return fmt.Errorf("Node Allocatable enforcement is not supported unless Cgroups Per QOS feature is turned on")
+		errors = append(errors, fmt.Errorf("Node Allocatable enforcement is not supported unless Cgroups Per QOS feature is turned on"))
 	}
 	if s.SystemCgroups != "" && s.CgroupRoot == "" {
-		return fmt.Errorf("invalid configuration: system container was specified and cgroup root was not specified")
+		errors = append(errors, fmt.Errorf("Invalid configuration: system container was specified and cgroup root was not specified"))
+	}
+	if s.CAdvisorPort < 0 || s.CAdvisorPort > 65535 {
```

Comment:
You may want to use `IsValidPortNum` (k8s.io/apimachinery/pkg/util/validation) instead for these port validations.